### ISSUE TITLE
Bun HTTP server crashes repeatedly in production.

### ISSUE THREAD

**Issue by songinnight** [oven-sh/bun#11053]
### How can we reproduce the crash?

I cannot reproduce it.
I've been using it since version 1.1.7 and it's crashing unexpectedly.
I switched server to Node.js from Bun now.

### JavaScript/TypeScript code that reproduces the crash?

I'm using PM2 Manager for application launching.

```shell
ecosystem.config.cjs

// ! PM2 CONFIG FOR BUN
module.exports = {
  apps: [
    {
      name: "api-sa",
      script: "main.mjs",
      instances: 1,
      exec_mode: "fork",
      merge_logs: false,
      autorestart: false,
      watch: false,
      listen_timeout: 10000,
      kill_timeout: 3000,
      env: {
        NODE_ENV: 'production',
      },
      env_production: {
        NODE_ENV: "production",
      },
      interpreter: "/home/ubuntu/.yarn/bin/bun",
    },
  ],
};



package.json

{
  "name": "super-admin-api",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node main.mjs"
  },
  "imports": {
    "#helper/*": "./helper/*",
    "#model/*": "./model/*",
    "#libs/*": "./libs/*"
  },
  "dependencies": {
    "@aws-sdk/client-ec2": "^3.574.0",
    "@aws-sdk/client-elastic-load-balancing": "^3.574.0",
    "@aws-sdk/client-elastic-load-balancing-v2": "^3.574.0",
    "@aws-sdk/credential-providers": "^3.574.0",
    "@aws-sdk/shared-ini-file-loader": "^3.374.0",
    "@fastify/cors": "^9.0.1",
    "@georgian83/domain-expiry2": "^1.3.0",
    "axios": "^1.6.8",
    "colors": "^1.4.0",
    "dotenv": "^16.4.5",
    "dotenv-expand": "^11.0.6",
    "excel4node": "^1.8.2",
    "execa": "8.0.1",
    "fastify": "^4.27.0",
    "http-status-codes": "^2.3.0",
    "jsonwebtoken": "^9.0.2",
    "moment-timezone": "^0.5.45",
    "otplib": "^12.0.1",
    "php-parser": "2.2.0",
    "php-unparser": "^0.2.9",
    "sqlite3": "^5.1.7"
  },
  "devDependencies": {
    "@types/node": "^20.12.10"
  },
  "module": "main.mjs"
}
```

### Relevant log output

```shell
0|api-sa   | Bun v1.1.8 (89d25807) Linux x64
0|api-sa   | Args: "/home/ubuntu/.yarn/bin/bun", "/server/super-admin-api/main.mjs"
0|api-sa   | Features: jsc Bun.stdin(2) dotenv fetch(3890) http_server transpiler_cache(6) tsconfig(4)
0|api-sa   | Builtins: "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" "node:crypto" "node:dns" "node:events" "node:fs" "node:fs/promises" "node:http" "node:https" "node:module" "node:net" "node:os" "node:path" "node:process" "node:stream" "node:string_decoder" "node:timers/promises" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" "node:worker_threads" "node:punycode" "node:http2" "node:diagnostics_channel"
0|api-sa   | Elapsed: 6578080ms | User: 95698ms | Sys: 13566ms
0|api-sa   | RSS: 0.78GB | Peak: 0.31GB | Commit: 0.78GB | Faults: 39
0|api-sa   |
0|api-sa   | panic(main thread): Segmentation fault at address 0x170000103A50
0|api-sa   | oh no: Bun has crashed. This indicates a bug in Bun, not your code.
0|api-sa   |
0|api-sa   | To send a redacted crash report to Bun's team,
0|api-sa   | please file a GitHub issue using the link below:
0|api-sa   |
0|api-sa   |  https://bun.report/1.1.8/la189d2580AqoihgF2/5ijE+xyQgj1Sk//Rky8kjDsj2r7Cojol6C0xk9+D2zzo5Csnhv5CA2gwLgl9gC
```

### Stack Trace (bun.report)

Bun v1.1.8 ([`89d2580`](https://github.com/oven-sh/bun/tree/89d25807fbf4488d3bfcb116484d0c2cf824f98c)) on linux x86_64 [AutoCommand]

Segmentation fault at address 0x170000103A50

- [`ld-temp.o:0`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/ld-temp.o#L0): `char32_t WTF::Unicode::next<(...)0, char8_t>`
- `??`
- `??`
- `??`
- [`event_loop.zig:1079`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/bun.js/event_loop.zig#L1079): `src.bun.js.event_loop.EventLoop.tickQueueWithCount__anon_361749`
- [`fifo.zig:242`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/deps/zig/lib/std/fifo.zig#L242): `src.bun.js.event_loop.EventLoop.tick`
- [`bindings.zig:4167`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/bun.js/bindings/bindings.zig#L4167): `src.bun.js.javascript.OpaqueWrap__anon_52755__struct_285184.callback`
- [`bindings.cpp:4749`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/bun.js/bindings/bindings.cpp#L4749): `JSC__JSPromise__rejectedPromiseValue`
- [`interpreter.zig:1267`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/shell/interpreter.zig#L1267): `src.shell.interpreter.Interpreter.initAndRunFromSource`
- [`cli.zig:1996`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/cli.zig#L1996): `src.cli.Command.maybeOpenWithBunJS`

**Comment by KilianB** [comment]
Maybe related to #11036?

**Comment by Jarred-Sumner** [comment]
If you still run into this, please provide some kind of a reproduction. We've fixed hundreds of bugs since Bun v1.1.8.

**Comment by cirospaciari** [comment]
@songinnight we have more fixes that will be in `v1.1.22` for now I would recommend `bun upgrade --canary`, the new release will happen soon.
